python/deltalake/table.py

```diff
@@ -575,7 +634,7 @@ def update(
             Dict[str, Union[int, float, str, datetime, bool, List[Any]]]
         ] = None,
         predicate: Optional[str] = None,
-        writer_properties: Optional[Dict[str, int]] = None,
+        writer_properties: Optional[WriterProperties] = None,
```

Comment:
This, and similar changes in other places will break the compatibility, is that intended?

Reply:
Yeah it will be a breaking change here, so we need to mark that in the release